astropy/io/ascii/ecsv.py

```diff
@@ -227,76 +48,29 @@ def write(self, lines):
-                ColumnOrderList object.  The Python 3 version insists on turning the
-                items() mapping into a list object and sorting, which results in
-                alphabetical order for the column keys.
-                """
-                value = []
-                node = yaml.MappingNode(tag, value, flow_style=flow_style)
-                if self.alias_key is not None:
-                    self.represented_objects[self.alias_key] = node
-                best_style = True
-                if hasattr(mapping, 'items'):
-                    mapping = mapping.items()
-                    if hasattr(mapping, 'sort'):
-                        mapping.sort()
-                    else:
-                        mapping = list(mapping)
-                        try:
-                            mapping = sorted(mapping)
-                        except TypeError:
-                            pass
-
-                for item_key, item_value in mapping:
-                    node_key = self.represent_data(item_key)
-                    node_value = self.represent_data(item_value)
-                    if not (isinstance(node_key, yaml.ScalarNode) and not node_key.style):
-                        best_style = False
-                    if not (isinstance(node_value, yaml.ScalarNode) and not node_value.style):
-                        best_style = False
-                    value.append((node_key, node_value))
-                if flow_style is None:
-                    if self.default_flow_style is not None:
-                        node.flow_style = self.default_flow_style
-                    else:
-                        node.flow_style = best_style
-                return node
-
-        TableDumper.add_representer(OrderedDict, _repr_odict)
-        TableDumper.add_representer(ColumnDict, _repr_column_dict)
-
         if self.splitter.delimiter not in DELIMITERS:
             raise ValueError('only space and comma are allowed for delimiter in ECVS format')
 
+        for col in self.cols:
+            if len(getattr(col, 'shape', ())) > 1:
+                raise ValueError("ECSV format does not support multidimensional column '{0}'"
+                                 .format(col_getattr(col, 'name')))
+
         # Now assemble the header dict that will be serialized by the YAML dumper
-        header = {}
+        header = {'cols': self.cols}
+
         if self.table_meta:
             header['meta'] = self.table_meta
 
-        header['datatype'] = [_get_col_attributes(col) for col in self.cols]
-
         # Set the delimiter only for the non-default option(s)
         if self.splitter.delimiter != ' ':
             header['delimiter'] = self.splitter.delimiter
 
-        header_yaml = yaml.dump(header, Dumper=TableDumper)
-        outs = ['%ECSV {0}'.format(ECSV_VERSION), '---']
-        outs.extend(header_yaml.splitlines())
+        header_yaml_lines = (['%ECSV {0}'.format(ECSV_VERSION),
```

Comment:
Bummer.  The `%` prefix is now required in the ECSV 1.0 standard in APE-6.

Regardless, this PR is just a refactoring and shouldn't be changing any functionality (for better or worse).  Are you OK with this @mdboom ?